### DISCUSSION TITLE
What am I doing wrong? Taking longer to run tests with Vitest than Jest!

### DISCUSSION THREAD

**Discussion by lukebennett88** [vitest-dev/vitest#1747] [Q&A]
I'm looking at switching the test framework for the design system I'm working on over to Vitest. We were using a pretty standard Jest/React Testing Library setup right now.
After switching over to Vitest, running the tests on 17 files, and 75 tests takes around 9 seconds on my local machine.
When I run the same tests using Jest, it takes a little over 4 seconds.

I've got a pretty standard looking vitest config:
```ts
/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  test: {
    globals: true,
    environment: 'jsdom',
    setupFiles: './test/setup-test-env.ts',
  },
});
```

The `setup-test-env.ts` file above is just this:

```ts
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
```

I'm not sure if this is related or not, but as you can see in my config above I have globals turned on. I uninstalled `@types/jest` from the codebase, but my editor is still picking up types for `describe`, `expect` etc. I ran `yarn why @types/jest` and it looks like the types are leaking out from `@testing-libarary/jest-dom`. I tried turning globals off, the testing library `jest-dom` package seems to be relying on them to be turned on as it fails with the error `ReferenceError: expect is not defined`. I'd like to turn globals off and import what I'm using, but unless I'm doing something wrong, it doesn't look like I can and still use React Testing Library.

**Comment by kamranayub** [comment]
Seeing very similar, literally running a single test file (with 3 tests) and it takes 13 seconds 😮 

```
Duration  13.54s (transform 216ms, setup 3.85s, collect 169ms, tests 47ms, environment 7.10s, prepare 1.52s)
```

I have a setup file like this:

```js
import { afterEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';

afterEach(() => {
  cleanup();
});
```

And my config is similar:

```js
test: {
    globals: true,
    environment: 'jsdom',
    setupFiles: '.testrc.js'
},
```

**Reply by sheremet-va** [reply]
Looks like `jsdom` takes a very long time to load. Try adding `test.pool: 'vmForks'` to you config to make it a bit faster.

**Reply by kamranayub** [reply]
@sheremet-va Thanks for the suggestion. Adding that doesn't make any material difference for my use case. I am running single files with 1-5 tests, so I'm always incurring some cold start cost; that's probably why.

```
 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  12:33:59
   Duration  14.02s (transform 95ms, setup 4.03s, collect 78ms, tests 12ms, environment 6.82s, prepare 9.34s)
```

**Reply by BBoehm** [reply]
> Looks like `jsdom` takes a very long time to load. Try adding `test.pool: 'vmForks'` to you config to make it a bit faster.

For us, switching to `pool: 'threads'` helped, speed increase around factor 7
